src/tools/kata-ctl/src/ops/env_ops.rs

```diff
@@ -0,0 +1,445 @@
+    let mem_info = sys_info::mem_info().context("get host memory information")?;
+    Ok(MemoryInfo {
+        total: mem_info.total,
+        available: mem_info.avail,
+        free: mem_info.free,
+    })
+}
+
+fn get_host_info() -> Result<HostInfo> {
+    let host_kernel_version = utils::get_kernel_version(utils::PROC_VERSION_FILE)?;
+    let (host_distro_name, host_distro_version) =
+        utils::get_distro_details(utils::OS_RELEASE, utils::OS_RELEASE_CLR)?;
+    let (cpu_vendor, cpu_model) = utils::get_generic_cpu_details(check::PROC_CPUINFO)?;
+
+    let host_distro = DistroInfo {
+        name: host_distro_name,
+        version: host_distro_version,
+    };
+
+    let cores: usize = std::thread::available_parallelism()
+        .context("get available parallelism")?
+        .into();
+
+    let host_cpu = CPUInfo {
+        vendor: cpu_vendor,
+        model: cpu_model,
+        cpus: cores,
+    };
+
+    let memory_info = get_memory_info()?;
+
+    let guest_protection = arch_specific::available_guest_protection()?.to_string();
+
+    let support_v_socks = utils::supports_vsocks(utils::VHOST_VSOCK_DEVICE)?;
+
+    Ok(HostInfo {
+        kernel: host_kernel_version,
+        architecture: String::from(std::env::consts::ARCH),
+        distro: host_distro,
+        cpu: host_cpu,
+        memory: memory_info,
+        available_guest_protection: guest_protection,
+        vm_container_capable: true, // TODO: implement
+        support_v_socks,
+    })
+}
+
+pub fn get_runtime_info(toml_config: &TomlConfig) -> RuntimeInfo {
+    let version = VersionInfo {
+        semver: String::from(version::VERSION),
+        commit: String::from(version::COMMIT),
+    };
+
+    let config_path = TomlConfig::get_default_config_file();
+    let mut toml_path = String::new();
+    if config_path.is_ok() {
+        let p = config_path.unwrap();
+        let path_str = p.to_str();
+        toml_path = match path_str {
+            Some(s) => String::from(s),
+            None => String::new(),
+        };
+    }
+
+    RuntimeInfo {
+        // TODO: Needs to be implemented
+        path: "TBD".to_string(),
+        version,
+        experimental: toml_config.runtime.experimental.clone(),
+        guest_se_linux_label: String::from("not implemented yet"),
+        debug: toml_config.runtime.debug,
+        trace: toml_config.runtime.enable_tracing,
+        disable_guest_seccomp: toml_config.runtime.disable_guest_seccomp,
+        disable_new_net_ns: toml_config.runtime.disable_new_netns,
+        sandbox_cgroup_only: toml_config.runtime.sandbox_cgroup_only,
+        static_sandbox_resource_mgmt: toml_config.runtime.static_sandbox_resource_mgmt,
+        config: RuntimeConfigInfo { path: toml_path },
+    }
+}
+
+pub fn get_agent_info(toml_config: &TomlConfig) -> Result<AgentInfo> {
+    let agent_config = toml_config
+        .agent
+        .get(&toml_config.runtime.agent_name)
+        .ok_or("could not find agent config in configuration")
```

Comment:
I hit this branch in testing:

```bash
$ grep '^\[agent' /etc/kata-containers/configuration.toml
[agent.kata]
sudo $(which kata-ctl)  env
ERROR: "could not find agent config in configuration"
```

However, this behaviour is incorrect due to an existing bug in the runtime-rs implementation - see https://github.com/kata-containers/kata-containers/issues/5954.

Reply:
Yes, came accross this issue as well. However I tested this PR with the dragonball config introduced in runtime-rs and it works with that.

Reply:
@jodh-intel I needed to implement this was as the agent field in the TomlConfig is a HashMap keyed on the agent name and not a single agent config structure:
https://github.com/kata-containers/kata-containers/blob/e90989b16b8ea8cfd0c7481f1c2406534d2b9699/src/libs/kata-types/src/config/mod.rs#L104
Similarly for the hypervisor field.  This would need to be resolved in how the toml is parsed and stored in the `kata-types` crate. Once it is resolved there, I can change it here. Wdyt?

Reply:
@jodh-intel I have addressed this in a following commit. For configs where agent_name and hypervisor_name are absent, I resort to using the first entry in the hashmap that is meant to store the agent and hypervisor configs by runtime-rs.